From 03cddb8c1dd8912e15b27e5a5c1cb8edcc9350b9 Mon Sep 17 00:00:00 2001
From: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Date: Fri, 7 Jul 2023 09:28:49 -0300
Subject: [PATCH] Update charts requirements to use local server

This change reverts openstack/openstack-helm commit [1] for charts that
the StarlingX OpenStack application is currently building.
That change was removing the helm-toolkit chart dependency from a local
server, since Helm v3 no longer supports "helm serve" [2], and pointing
it to a given openstack-helm-infra directory in which the helm-toolkit
chart should be placed.
The stx-openstack application does not require this change while it is
relying on chartmuseum for serving charts locally [3].
Instead of changing our build instructions and our custom charts,
including charts for other repositories [4], we simply reverts
openstack-helm requirements to use local server again.

[1] c20c1e4400f5935adf0afd0c65bef2bb12af598b
[2] https://helm.sh/docs/topics/v2_v3_migration/
[3] https://opendev.org/starlingx/openstack-armada-app/src/branch/r/
    stx.8.0/stx-openstack-helm-fluxcd/debian/deb_folder/control#L6
[4] https://opendev.org/starlingx/openstack-armada-app/src/branch/r/
    stx.8.0/stx-openstack-helm-fluxcd/debian/meta_data.yaml#L7

Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Change-Id: Id2ab4adabb21201da229e4242fe06c1ba1bfd463
---
 aodh/requirements.yaml       | 2 +-
 barbican/requirements.yaml   | 2 +-
 ceilometer/requirements.yaml | 2 +-
 cinder/requirements.yaml     | 2 +-
 glance/requirements.yaml     | 2 +-
 heat/requirements.yaml       | 2 +-
 horizon/requirements.yaml    | 2 +-
 ironic/requirements.yaml     | 2 +-
 keystone/requirements.yaml   | 2 +-
 magnum/requirements.yaml     | 2 +-
 neutron/requirements.yaml    | 2 +-
 nova/requirements.yaml       | 2 +-
 placement/requirements.yaml  | 2 +-
 13 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/aodh/requirements.yaml b/aodh/requirements.yaml
index 36f1a6e0..fbba94ae 100644
--- a/aodh/requirements.yaml
+++ b/aodh/requirements.yaml
@@ -14,5 +14,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/barbican/requirements.yaml b/barbican/requirements.yaml
index 4124d014..432e28c1 100644
--- a/barbican/requirements.yaml
+++ b/barbican/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/ceilometer/requirements.yaml b/ceilometer/requirements.yaml
index 4124d014..432e28c1 100644
--- a/ceilometer/requirements.yaml
+++ b/ceilometer/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/cinder/requirements.yaml b/cinder/requirements.yaml
index 4124d014..432e28c1 100644
--- a/cinder/requirements.yaml
+++ b/cinder/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/glance/requirements.yaml b/glance/requirements.yaml
index 4124d014..432e28c1 100644
--- a/glance/requirements.yaml
+++ b/glance/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/heat/requirements.yaml b/heat/requirements.yaml
index 4124d014..432e28c1 100644
--- a/heat/requirements.yaml
+++ b/heat/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/horizon/requirements.yaml b/horizon/requirements.yaml
index 4124d014..432e28c1 100644
--- a/horizon/requirements.yaml
+++ b/horizon/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/ironic/requirements.yaml b/ironic/requirements.yaml
index 4124d014..432e28c1 100644
--- a/ironic/requirements.yaml
+++ b/ironic/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/keystone/requirements.yaml b/keystone/requirements.yaml
index 4124d014..432e28c1 100644
--- a/keystone/requirements.yaml
+++ b/keystone/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/magnum/requirements.yaml b/magnum/requirements.yaml
index 4124d014..432e28c1 100644
--- a/magnum/requirements.yaml
+++ b/magnum/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/neutron/requirements.yaml b/neutron/requirements.yaml
index 4124d014..432e28c1 100644
--- a/neutron/requirements.yaml
+++ b/neutron/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/nova/requirements.yaml b/nova/requirements.yaml
index 4124d014..432e28c1 100644
--- a/nova/requirements.yaml
+++ b/nova/requirements.yaml
@@ -12,5 +12,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
diff --git a/placement/requirements.yaml b/placement/requirements.yaml
index 639dab0a..7efb17a3 100644
--- a/placement/requirements.yaml
+++ b/placement/requirements.yaml
@@ -14,5 +14,5 @@
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../../openstack-helm-infra/helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
-- 
2.25.1

